src/components/Table/PaginatedTable/Pagination.jsx

```diff
@@ -35,32 +33,69 @@ const getPages = ({
       {i + 1}
     </FlatButton>);
   }
-  pages.push(<FlatButton className={styles.page} onClick={() => setCurrentPage(numPages - 1)}>...{numPages}</FlatButton>);
   return pages;
 };
 
-const Pagination = ({
+const PaginationBottom = ({
   currentPage,
   nextPage,
   prevPage,
   setCurrentPage,
   numPages,
-}) => (numPages > 1 ? (
-  <div className={styles.container}>
-    <FlatButton className={styles.arrow} onClick={currentPage > 0 ? prevPage : () => {}}><Prev className={styles.arrow} /></FlatButton>
-    {getPages({ currentPage, numPages, setCurrentPage })}
-    <FlatButton className={styles.arrow} onClick={currentPage < (numPages - 1) ? nextPage : () => {}}><Next className={styles.arrow} /></FlatButton>
+  pageLength,
+  length,
+  place,
+}) => numPages > 1 && (
+  <div className={place === 'top' ? styles.containerTop : styles.container}>
+    <div className={place === 'top' ? styles.paginationTop : styles.pagination}>
+      {currentPage > 0 &&
+        <FlatButton className={styles.page} onClick={() => setCurrentPage(0)}>
+          {strings.pagination_first}
+        </FlatButton>
+      }
+      <div className={styles.pages}>
+        {currentPage > 0 &&
+          <FlatButton className={styles.arrow} onClick={currentPage > 0 ? prevPage : () => {}}>
+            <Prev className={styles.arrow} />
+          </FlatButton>
+        }
+        {currentPage > 2 && numPages > 2 &&
+          <FlatButton disabled className={styles.currentPage}>
+            ...
+          </FlatButton>
+        }
+        {getPages({ currentPage, numPages, setCurrentPage })}
+        {numPages > currentPage + 3 &&
+          <FlatButton disabled className={styles.currentPage}>
+            ...
+          </FlatButton>
+        }
+        {currentPage < numPages - 1 &&
+          <FlatButton className={styles.arrow} onClick={currentPage < (numPages - 1) ? nextPage : () => {}}>
+            <Next className={styles.arrow} />
+          </FlatButton>
+        }
+      </div>
+      {currentPage < numPages - 1 &&
+        <FlatButton className={styles.page} onClick={() => setCurrentPage(numPages - 1)}>
+          {strings.pagination_last}
+        </FlatButton>
+      }
+    </div>
+    {place === 'bot' &&
+      <div className={styles.info}>
+        {(pageLength * currentPage).toLocaleString('ru-RU')}
```

Comment:
I chose 'ru' because imo it's the most understandable format for everyone.
If it's wrong we can change it latter, when language selector appears 😛